src/rust/ensogl/src/display/navigation/navigator/events.rs

```diff
@@ -290,13 +290,11 @@ impl NavigatorEvents {
             if let Some(data) = data.upgrade() {
                 let position = Vector2::new(event.offset_x() as f32, event.offset_y() as f32);
                 data.set_mouse_position(position);
-                let mut movement = data.mouse_position() - data.last_mouse_position();
-                movement.x = -movement.x;
+                let movement = data.last_mouse_position() - data.mouse_position();
```

Comment:
Something is wrong here. Movement = end position - source position. Here it is negative. Why? It needs to be explained or a variable should be named better. But yeah, lets start with the question why?

Reply:
@wdanilo yes, it only makes sense if you read the diff from the previous version. I provided a cleaner implementation